src/vma/sock/sockinfo_tcp.cpp

```diff
@@ -3533,18 +3533,17 @@ int sockinfo_tcp::setsockopt(int __level, int __optname,
 			// TODO handle RX side
 			si_tcp_logdbg("(SO_BINDTODEVICE) interface=%s", (char*)__optval);
 			break;
-		case SO_MAX_PACING_RATE:
-			val = *(int *)__optval;
-			lock_tcp_con();
-			if (-1 == modify_ratelimit(m_p_connected_dst_entry, val)) {
-                                si_tcp_logdbg("error setting setsockopt SO_MAX_PACING_RATE: %d bytes/second ", val);
-			} else {
-	                        si_tcp_logdbg("setsockopt SO_MAX_PACING_RATE: %d bytes/second ", val);
+		case SO_MAX_PACING_RATE: {
+			val = *(uint32_t *)__optval;
+			auto_unlocker t_lock(m_tcp_con_lock);
```

Comment:
@ilansmith the lock is unlocked at the end of the brackets in the break statment (line 3545). so it will be held the same time.
about the style the auto lock was added to VMA after the setsockopt was added thats why it wasn't used in it. I prefer using the new style witch is less buggy and taked ledd code lines then the old style.
changing the all the locks in this function is an option but i will not do it during this commit due to lack of time.
please decide